### PR TITLE
Map template, plants, turfs, floors, gravity, atmos

### PR DESCRIPTION
**Pull request by ma44**
## About The Pull Request
Introduces a basic map, plants, turfs, floors, gravity and planetary atmos

## Why It's Good For The Game
We can do more things:tm:

## Changelog
:cl:
add: A basic map with a outer border of matrix walls has been placed
add: Procedural generated plants
add: Gravity!
add: Planetary atmos to filter stuff
add: Defined turfs for map makers (floors and walls)
add: Ports plants
/:cl:

### CODE REVIEW

**Comment by ma44** [comment]
fuck

File: code/__HELPERS/icon_smoothing.dm [hunk 1]
```diff
@@ -117,7 +118,12 @@
 		return
 	if(QDELETED(A))
 		return
-	if(A.smooth & (SMOOTH_TRUE | SMOOTH_MORE))
+
+	if(A.smooth & SMOOTH_OLD)
+		A:recalculate_junction()
```
Comment (optimumtact) [inline code/__HELPERS/icon_smoothing.dm]: take a look at THIS

File: code/controllers/subsystem/mapping.dm [hunk 2]
```diff
@@ -248,8 +248,9 @@ SUBSYSTEM_DEF(mapping)
 	// load mining
 	if(config.minetype == "lavaland")
 		LoadGroup(FailedZs, "Lavaland", "map_files/Mining", "Lavaland.dmm", default_traits = ZTRAITS_LAVALAND)
-	else if (!isnull(config.minetype))
-		INIT_ANNOUNCE("WARNING: An unknown minetype '[config.minetype]' was set! This is being ignored! Update the maploader code!")
+	//TG CLAW: NO MORE LAVALAND
+	//else if (!isnull(config.minetype))
```
Comment (optimumtact) [inline code/controllers/subsystem/mapping.dm]: AND THIS

File: fallout/flora/wasteplants.dm [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	icon = 'fallout/icons/flora/wastelandflora.dmi'
+	anchored = 1
+	density = 0
+	var/has_prod = TRUE
+	var/obj/item/reagent_containers/food/snacks/grown/produce = null //If only we used ACTUAL plants
+
+/obj/structure/flora/wasteplant/attack_hand(mob/user)
+	if(has_prod)
+		user.put_in_hands(new produce)
+		user << "<span class='notice'>You take [produce] from [src].</span>"
```
Comment (optimumtact) [inline fallout/flora/wasteplants.dm]: oh no no nono

**Comment by ma44** [comment]
Tfw code

**Comment by cacogen** [comment]
it's always sunny in the dale

**Pull request closed without merging**